### PR TITLE
Fix newBlockFilter + getBlockByHash

### PR DESCRIPTION
**Pull request by eduardonunesp**
- [x] Identify the issue
- [x] Create a test for the issue

This PR also improve the `filter` functionality and `indexed` filter functionality

### CODE REVIEW

File: src/crypto-utils.ts [hunk 1]
```diff
@@ -14,6 +14,17 @@ export function hexToNumber(hex: string): number {
   return parseInt(hex, 16)
 }
 
+export function hexAddrToUint8Array(hexAddr: string): Uint8Array {
```
Comment (enlight) [inline src/crypto-utils.ts]: This is no longer necessary, lets remove it.

File: src/tests/helpers.ts [hunk 2]
```diff
@@ -37,3 +37,7 @@ export function createTestHttpWSClient(): Client {
   })
   return new Client('default', writer, reader)
 }
+
+export function timeout(ms: number) {
```
Comment (enlight) [inline src/tests/helpers.ts]: I'd prefer `export function waitForMillisecondsAsync(ms: number): Promise<void>`

File: src/client.ts [hunk 3]
```diff
@@ -594,6 +621,8 @@ export class Client extends EventEmitter {
       const eventArgs: IClientErrorEventArgs = { kind: ClientEvent.Error, url, error }
       this.emit(ClientEvent.Error, eventArgs)
     } else if (result) {
+      console.log('event', event.id, result)
```
Comment (enlight) [inline src/client.ts]: If you want to leave this in change it to use `debug` instead of `console.log`.

Comment (eduardonunesp) [reply src/client.ts]: Ops I forgot this one

File: src/crypto-utils.ts (hunk 1 above)
Comment (eduardonunesp) [reply src/crypto-utils.ts]: Removed

**Pull request merged**